task/backend/coordinator/coordinator.go

```diff
@@ -126,7 +131,14 @@ func (c *Coordinator) UpdateTask(ctx context.Context, id platform.ID, upd platfo
 	}
 
 	// If enabling the task, claim it after modifying the script.
-	if task.Status == string(backend.TaskActive) {
+	if task.Status != oldTask.Status && task.Status == string(backend.TaskActive) {
+		// don't catch up on all the missed task runs while disabled
+		newLatestCompleted := time.Now().UTC().Format(time.RFC3339)
```

Comment:
The other uses of time.Now in `influxdb/task` and its subpackages are related to timestamps for run logs, setting created/updated at, and managing the scheduler's ticker.

Instead of calling time.Now() here, add a new method to the Scheduler interface that exposes the scheduler's current time, and use that. Then you can avoid the hardcoded time.Sleep in the new test.